Essentials/src/main/java/com/earth2me/essentials/config/serializers/LocationTypeSerializer.java

```diff
@@ -0,0 +1,66 @@
+package com.earth2me.essentials.config.serializers;
+
+import org.bukkit.Bukkit;
+import org.bukkit.Location;
+import org.bukkit.World;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.spongepowered.configurate.ConfigurationNode;
+import org.spongepowered.configurate.serialize.SerializationException;
+import org.spongepowered.configurate.serialize.TypeSerializer;
+
+import java.lang.reflect.Type;
+import java.util.UUID;
+
+/**
+ * A Configurate type serializer for {@link Location}s.
+ *
+ * Locations with a null or empty world will be considered invalid.
+ */
+public class LocationTypeSerializer implements TypeSerializer<Location> {
+    @Override
+    public Location deserialize(Type type, ConfigurationNode node) throws SerializationException {
+        final String worldValue = node.node("world").getString();
+        if (worldValue == null || worldValue.isEmpty()) {
+            throw new SerializationException("No world value present!");
+        }
+
+        World world = null;
+
+        try {
+            final UUID worldId = UUID.fromString(worldValue);
+            world = Bukkit.getWorld(worldId);
```

Comment:
this is bad --- what if a location is stored for an unloaded world?

probably a pain to fix and I've seen this in iirc `Jails` as well, so can be handled in a later pass